### PR TITLE
Requests

### PR DESCRIPTION
**Pull request by dborski**
## Description
Completes User Story #10 
## Contributors
Derek Borski - @dborski 
Max Mitrani - @Lithnotep 
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Notes
- This PR creates functionality for two different endpoints:
  - api/v1/teachers
    - Get all teachers
    - Create a teacher
  - api1/v1/teachers/:teacher_id
    - Request specific teacher information by id

- Add factories:
  - Teacher
  - Student
## RSpec Results
- 100%

### CODE REVIEW

File: api/serializers.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from rest_framework import serializers
+
+from api.models import Teacher
+
+class TeacherSerializer(serializers.ModelSerializer):
+  class Meta:
+    model = Teacher
+    fields = (
+      'id', 'first_name', 'last_name'
+    )
```
Comment (Lithnotep) [inline api/serializers.py:10]: First serializer is looking good.

File: api/tests/factories.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+class StudentFactory(factory.django.DjangoModelFactory):
+  class Meta:
+    model = models.Student
+
+  first_name = factory.Sequence(lambda n: 'student%dFirst' % n)
+  last_name = factory.Sequence(lambda n: 'student%dLast' % n)
+  age = fake.random_digit()
+  teacher = factory.SubFactory(TeacherFactory)
+
```
Comment (Lithnotep) [inline api/tests/factories.py:24]: thank you for setting up some factories

File: api/views.py [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+
+  
+class TeacherDetail(generics.RetrieveAPIView):
+  queryset = Teacher.objects.all()
+  serializer_class = TeacherSerializer
+
+  # def get(self, request, pk):
+  #   teacher = Teacher.objects.get(pk=pk)
+
+  #   serializer = TeacherSerializer(teacher)
```
Comment (Lithnotep) [inline api/views.py:22]: "Houston we are restful"

**Review by Lithnotep — APPROVED** [review]

File: api/tests/factories.py (hunk 2 above)
Comment (melatran) [reply api/tests/factories.py:24]: glad to use factories again.

**Pull request merged**